s3fs/core.py

```diff
@@ -142,7 +169,10 @@ def _ls(self, path, refresh=False):
                     f['Size'] = 0
                     del f['Name']
             else:
-                files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                try:
+                    files = self.s3.list_objects(Bucket=bucket).get('Contents', [])
+                except ClientError:
+                    files = []
```

Comment:
Under what conditions does the `ClientError` get raised?

Is there a cheap way to determine the existence of a bucket?

When listing out keys in the bucket if there are no matches perhaps this means that the implied directory does not exist?